src/app/shared/services/schema-keys-store.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import { fromJS, OrderedSet } from 'immutable';
+
+@Injectable()
+export class SchemaKeysStoreService {
+
+  private schemaSeparator = '/';
+  public keyStoreMap: { [path: string]: OrderedSet<string> } = {};
+
+  constructor() { }
+
+  public buildSchemaKeyStore(schema: {}) {
+    this.buildSchemaKeyStoreRecursively('', schema);
+  }
+
+  private buildSchemaKeyStoreRecursively(path: string, schema: {}) {
+
+    if (schema['type'] === 'object') {
```

Comment:
There seems to be quite a lot of repetition between the two branches (`object` and `array`). Would be good combine them, and for one use `schema['properties']` and for the other one `schema['items']['properties']`